File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.py
def solution(array):
    number = [0] * 1001
    for a in array:
        number[a] += 1
    
    ans = max(number)
    cnt = number.count(ans)
    if cnt == 1: 
        answer = number.index(ans)   
    else:
        answer = -1
    
    return answer

File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[] array) {
        int[] numbers = new int[1001];
        for (int i = 0; i < array.length; i++) {
            numbers[array[i]] += 1;
        }
                
        int max = 0;
        int idx = 0;
        for (int i = 0; i <= 1000; i++) {
            if (max < numbers[i]) {
                max = numbers[i];
                idx = i;
            }
        }
        
        int cnt = 0;
        for (int i = 0; i <= 1000; i++) {
            if (cnt > 1) {
                break;
            } else if (numbers[i] == max) {
                cnt += 1;
            }
        }
        
        int answer = 0;
        if (cnt > 1) {
            answer = -1;
        } else {
            answer = idx;
        }
        
        return answer;
    }
}